### PR TITLE
Clock.py types.

### PR DESCRIPTION
**Pull request by InvincibleRMC**
Add types to clock.py. Waiting on #1240.

### CODE REVIEW

**Review by sloretz — APPROVED** [review]
LGTM with green CI!

**Comment by sloretz** [comment]
CI ([repos file](https://gist.githubusercontent.com/sloretz/fd47fd36aa361dd10d7fe016a17048ba/raw/ab0176c8c6f49fd9be4bb920077b58386ba027b7/ros2.repos) build: `--packages-up-to rclpy` test: `--packages-select rclpy`)
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/20763/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/15062/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/21444/)

**Comment by sloretz** [comment]
Looks like there's a test failure with the `new_post_callback` change. Mind taking a look?

**Comment by InvincibleRMC** [comment]
> Looks like there's a test failure with the `new_post_callback` change. Mind taking a look?

Ah I found the mistake. I believe I have fixed it.

**Review by sloretz — APPROVED** [review]

**Comment by sloretz** [comment]
CI rebuild
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/20767/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/15065/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/21448/)

**Review by fujitatomoya — APPROVED** [review]
lgtm with green CI

File: rclpy/rclpy/destroyable.py [hunk 1]
```diff
@@ -17,7 +17,6 @@
 
 
 class DestroyableType(Protocol):
-
```
Comment (fujitatomoya) [inline rclpy/rclpy/destroyable.py:20]: https://ci.ros2.org/job/ci_linux/20767/testReport/ needs this line, and this is unrelated change.

Comment (InvincibleRMC) [reply rclpy/rclpy/destroyable.py]: done

**Comment by sloretz** [comment]
CI rebuild
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/20768/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/15066/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/21449/)

**Comment by InvincibleRMC** [comment]
This good to be merged in?

**Pull request merged**